include/fix8/thread.hpp

```diff
@@ -85,10 +85,11 @@ class _f8_threadcore
 #if (FIX8_THREAD_SYSTEM == FIX8_THREAD_PTHREAD)
 		return pthread_create(&_tid, &_attr, _run<T>, sub);
 #elif (FIX8_THREAD_SYSTEM == FIX8_THREAD_STDTHREAD)
-		if (_thread.get()) {
+		if (std::thread* thread_ptr = _thread_ptr.exchange(nullptr))
+		{
 			throw std::logic_error("_f8_threadcore::_start called on an active thread");
 		}
-		_thread.reset(new std::thread(_run<T>, sub));
+		_thread_ptr.store(new std::thread(_run<T>, sub));
```

Comment:
there is a race condition here.
you should do a swap and validate on value returned by swap.

Reply:
It's an error to call _start with a running thread, that std::logic_error is thrown if you do.

Reply:
i said that there is a race condition here. its not properly handled multi threaded code.
thread a) invokes .exchange(nullptr) if false because _thread_ptr was null, steps over the if statement and is interrupted before invoking store()
thread b) invokes.exchange(nullptr) if false again because thread a) didnt get to store call.

Hence this is not proper use of atomics in this case. there is a race condition here.

Reply:
It's an error to call this method, _start(), if there's a running thread.
If that does happen, we get a value in thread_ptr, and throw a logic_error.
The normal case is we create a running thread and exit.

Reply:
I am very well aware of whats this function is trying to achieve.
I am saying that the use of atomics is clearly wrong here.

exchange(nullptr)
store(..) 
has race condition that another thread could enter in the meantime and do the replacement.

please change the code above into:
```
		if (std::thread* thread_ptr = _thread_ptr.exchange(new std::thread(_run<T>, sub)))
		{
                        // we purposely do not delete thread_ptr here as joining on that leads to deadlock.
			throw std::logic_error("_f8_threadcore::_start called on an active thread");
		}
```

and this will be fine.

Reply:
Change applied.